addons/mail/models/mail_channel.py

```diff
@@ -778,7 +778,7 @@ def _execute_channel_pin(self, pinned=False):
             [('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', self.id)])
         self.env['bus.bus'].sendone((self._cr.dbname, 'res.partner', self.env.user.partner_id.id), self.channel_info('unsubscribe' if not pinned else False)[0])
         if channel_partners:
-            channel_partners.write({'is_pinned': pinned})
+            channel_partners.filtered(lambda x: x.is_pinned != pinned).write({'is_pinned': pinned})
```

Comment:
Could be inlined directly in hereabove search, isn't it ?

Reply:
@tde-banana-odoo it is modified